raft.go

```diff
@@ -927,6 +940,23 @@ func (r *Raft) appendEntries(rpc RPC, a *AppendEntriesRequest) {
 	return
 }
 
+// checkAndProcessConfigurationLog takes a log entry and updates the latest
+// configuration if the entry results in a new configuration. This must only be
+// called from the main thread.
```

Comment:
Given what it does I might make it just a plain old function that takes the state in though - will be easier to test.